packages/ui/src/components/Slider/SingleSlider.tsx

```diff
@@ -0,0 +1,348 @@
+    }
+    window.addEventListener('resize', setWidthResize)
+
+    return () => {
+      window.removeEventListener('resize', setWidthResize)
+    }
+  }, [])
+
+  const handleChange = useCallback(
+    (newValue: number) => {
+      if (possibleValues) {
+        // Custom scale
+        const optionLabel = possibleValues[newValue ?? min]
+        setCustomValue(optionLabel)
+        setValuesToShow(newValue)
+      } else {
+        setValuesToShow(newValue)
+      }
+    },
+    [min, possibleValues],
+  )
+
+  const getBackgroundSize = useMemo(
+    () => ({
+      backgroundSize: `${((computedValue - min) * 100) / (max - min)}% 100%`,
+    }),
+    [computedValue, min, max],
+  )
+
+  const styledValue = (valueNumber: string | number | null) =>
+    input && (!options || options === true) && !possibleValues ? (
+      <StyledNumberInput
+        value={
+          typeof valueNumber === 'string'
+            ? parseFloat(valueNumber)
+            : valueNumber
+        }
+        size="small"
+        min={min}
+        aria-label="input"
+        max={max}
+        step={step}
+        controls={false}
+        data-testid={dataTestId ? `${dataTestId}-input` : 'slider-input'}
+        unit={typeof suffix === 'string' ? suffix : unit}
+        onChange={newVal => {
+          if (newVal) {
+            setValuesToShow(newVal)
+          } else setValuesToShow(null)
+        }}
+        onBlur={event => {
+          if (!event.target.value) setValuesToShow(min)
+        }}
+      />
+    ) : (
+      <StyledTextValue
+        as="span"
+        variant="bodySmall"
+        sentiment="neutral"
+        placement={direction === 'column' ? 'right' : 'center'}
+        double={false}
+        isColumn={direction === 'column'}
+        data-testid={dataTestId ? `${dataTestId}-value` : 'slider-value'}
+      >
+        {prefix}
+        {valueNumber}
+        {suffix ?? unit}
+      </StyledTextValue>
+    )
+
+  const tooltipText = useMemo(() => {
+    if (tooltip === true) {
+      return computedValue
+    }
+    if (tooltip) {
+      return tooltip
+    }
+
+    return null
+  }, [tooltip, computedValue])
+
+  const placementTooltip =
+    ((computedValue - min) / (max - min)) * (sliderWidth - THUMB_SIZE) +
+    THUMB_SIZE / 2 -
+    sliderWidth / 2
+
+  return (
+    <Stack gap={1} direction={direction} justifyContent="left">
+      {label ? (
+        <Stack justifyContent="space-between" direction="row">
+          <Stack gap={0.5} direction="row">
+            <Text
+              as="label"
+              variant="bodyStrong"
+              htmlFor={finalId}
+              placement="left"
+            >
+              {label}
+            </Text>
+            {required ? (
+              <Icon name="asterisk" sentiment="danger" size={8} />
+            ) : null}
+          </Stack>
+
+          {direction === 'column'
+            ? styledValue(customValue ?? valueToShow)
+            : null}
+        </Stack>
+      ) : null}
```

Comment:
label is same in both single and double, couldn't it be unified in a`Label.tsx` component?